test/gtest/helper.h

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2013-present Barefoot Networks, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include <string>
+
+/* preprocessing by prepending the content of core.p4 to test program */
+std::string with_core_p4 (std::string pgm) {
```

Comment:
Again, since you don't mutate `pgm`, it's probably better to make it a `const std::string&`.

I agree it'd be best to read `core.p4` from the file once when the GTest executable starts (so in `main()`). We can then reuse it for every invocation of `with_core_p4`. If that's not quick to do, though, I think it'd be fine to do this now and file a followup issue to fix that later.